lib/ffmpeg/encoding_options.rb

```diff
@@ -38,8 +101,27 @@ def height
       self[:resolution].split("x").last.to_i rescue nil
     end
 
+    def attempt_self_call(key, value)
+      if value
+        if supports_option_public?(key)
+          public_send("convert_#{key}", value)
+        elsif supports_option_private?(key)
+          send("convert_#{key}", value)
+        end
+      end
+    end
+
+    def convert_inputs(values)
+      "-i #{values.join(' -i ')}"
+    end
+
     private
-    def supports_option?(option)
+    def supports_option_public?(option)
+      option = RUBY_VERSION < "1.9" ? "convert_#{option}" : "convert_#{option}".to_sym
```

Comment:
pardon my lack of knowledge here, but are we still running versions of this less than ruby 1.9?

Reply:
This is because the library is "generic" and forked from the original [Streamio](https://github.com/streamio/streamio-ffmpeg) version 😅

Reply:
I'll be honest, I'm not sure if anything else in here supports < 1.9, but I do know it _definitely supports_ 3.2 🚀 (though also questionable how much it supports < 3.2)

Reply:
ahhh there it is, makes sense